### DISCUSSION TITLE
Regular expression for the password generator

### DISCUSSION THREAD

**Discussion by w15eacre** [keepassxreboot/keepassxc#11584] [Ideas]
The password generator allows users to provide a sequence (also "**Also Choose from**" or "**Do not include**"). The first option is clear: the generator may use your specified sequence. The "**Do not include**" option, however, also lets users specify characters to exclude, but it becomes inconvenient when you need to exclude a large number of symbols.

For example, if I need to generate a hex password in lowercase, the "hex" button doesn’t work properly because it only excludes uppercase symbols. A solution for this could be using a **regular expression**. To exclude non-hex symbols, you can use: **[^a-fA-F0-9]**.

I can share a PoC if you need it.

**Comment by droidmonkey** [comment]
We are not making changes to the password generator anymore. You are welcome to implement this for yourself but we are unlikely to merge it.

**Reply by w15eacre** [reply]
Ok, could you say what the reason for that is?

**Reply by droidmonkey** [reply]
For one this is an anti-pattern to generating passwords, for two this introduces more complexity and possibly more input elements on the generator which is already rather complex.
